lib/fog/aws/requests/compute/attach_volume.rb

```diff
@@ -53,6 +53,8 @@ def attach_volume(instance_id, volume_id, device)
                 'status'      => 'attaching',
                 'volumeId'    => volume_id
               }
+              # Insert new mapping information into the instance
+              instance['blockDeviceMapping'] << data
```

Comment:
I believe `data` needs to be transformed into [the block device mapping type](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_BlockDeviceMapping.html)